ptvsd/wrapper.py

```diff
@@ -1908,12 +1955,21 @@ def on_pydevd_thread_suspend(self, seq, args):
                 pydevd_comm.CMD_STEP_INTO,
                 pydevd_comm.CMD_STEP_OVER,
                 pydevd_comm.CMD_STEP_RETURN,
+                pydevd_comm.CMD_STEP_INTO_MY_CODE,
         }
         EXCEPTION_REASONS = {
             pydevd_comm.CMD_STEP_CAUGHT_EXCEPTION,
             pydevd_comm.CMD_ADD_EXCEPTION_BREAK
         }
 
+        xframes = list(xml.thread.frame)
+        xframe = xframes[0]
+        filepath = unquote(xframe['file'])
+        if reason in STEP_REASONS or reason in EXCEPTION_REASONS:
+            if not self._should_debug(filepath):
```

Comment:
Is this really needed? In which cases do you expect `ptvsd` to do this auto-run which `pydevd` is not doing out of the box? 

-- the cases not supported should probably be bugs against `pydevd` as doing this auto-run is much slower than doing the skip on `pydevd` directly.

Reply:
This is needed for Step over and Step out. For example: 
```python
import unittest

class MyTest(unittest.TestCase):
    def test_0(self):
        pass

if __name__ == '__main__':
    unittest.main()
```

Set breakpoint on line with `pass`. Then do step over, it will enter into `case.py` in `unittest`.

Reply:
Ok, got it... this really needs a `just my code` flag globally in `pydevd` or 2 new commands (to step over my code and step return to my code) -- can you create an issue for that (and mark as `upstream-pydevd`)?

Also, given this auto-run in the example below:

```
import unittest

class MyTest(unittest.TestCase):
    def test_0(self):
        pass  # break here

if __name__ == '__main__':
    try:
        unittest.main()
    except SystemExit:
        print('finish')
```

a `step over`/`step out` should end up in `print('finish')` right? -- I think that with the current code it'd finish the program as you're just making it run and not stop at the next possible location as would be expected (or maybe I missed something?) -- if that's the case, I think the auto-run should be disabled (and the support for step over/step return would have to wait for that feature to be implemented in the `pydevd` side).

Reply:
You are correct, currently we just continue and will not end up in `print('finish')`. I will remove the auto run code.

Reply:
Created a issue for step over/step return just my code : #477